src/compiler/crystal/syntax/lexer.cr

```diff
@@ -2952,6 +2957,12 @@ module Crystal
       is_slash_r
     end
 
+    def char_sequence?(*tokens, column_increment : Bool = true)
```

Comment:
I'm not going to comment on if the abstraction make sense or not, that is for people that know the lexer better than me. 

However, while the lexer is not the most performance critical part of the compiler chain, this does add an array allocation to the parse of every keyword. Is there any measurable difference in parsing time from this?

Reply:
Where would the array allocation be?

Reply:
@yxhuvud IIRC no array allocations here, just tuples.

Reply:
Oh, it is generating a new version of the method for each *-size? Ah, ok.